docs/modeling/relationships.rst

```diff
@@ -0,0 +1,236 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The most common pattern for relationships is to have navigation properties defined on both ends of the relationship and a foreign key property defined in dependent entity class.
+ - If a pair of navigation properties is found between two types, then they will be configured as inverse navigation properties of the same relationship.
+ - If the dependent entity contains a property named ``<primary key property name>``, ``<navigation property name><primary key property name>``, or ``<principal entity name><primary key property name>`` then it will be configured as the foreign key.
+
+.. literalinclude:: configuring/sample/EFModeling.Conventions/Samples/RelationshipFull.cs
+       :language: c#
+       :lines: 12-28
+       :emphasize-lines: 6, 15-16
+       :linenos:
+
+.. caution::
+    If there are multiple navigation properties defined between two types (i.e. more than one distinct pair of navigations that point to each other), then no relationships will be created by convention and you will need to manually configure them to identify how the navigation properties pair up.
+
+No Foreign Key Property
+^^^^^^^^^^^^^^^^^^^^^^^
+
+While it is recommended to have a foreign key property defined in the dependent entity class, it is not required. If no foreign key property is found, a shadow foreign key property will be introduced with the name ``<principal type name><principal key property name>`` (see :doc:`shadow-properties` for more information).
+
+.. literalinclude:: configuring/sample/EFModeling.Conventions/Samples/RelationshipNoForeignKey.cs
+        :language: c#
+        :lines: 12-27
+        :emphasize-lines: 6, 15
+        :linenos:
+
+Single Navigation Property
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Including just one navigation property (no inverse navigation, and no foreign key property) is enough to have a relationship defined by convention. You can also have a single navigation property and a foreign key property.
+
+.. literalinclude:: configuring/sample/EFModeling.Conventions/Samples/RelationshipOneNavigation.cs
+        :language: c#
+        :lines: 12-25
+        :emphasize-lines: 6
+        :linenos:
+
+Data Annotations
+----------------
+
+There are two data annotations that can be used to configure relationships, ``[ForeignKey]`` and ``[InverseProperty]``.
+
+[ForeignKey]
+^^^^^^^^^^^^
+
+You can use the Data Annotations to configure which property should be used as the foreign key property for a given relationship. This is typically done when the foreign key property is not discovered by convention.
+
+.. literalinclude:: configuring/sample/EFModeling.Configuring.DataAnnotations/Samples/RelationshipForeignKey.cs
+        :language: c#
+        :lines: 13-31
+        :emphasize-lines: 17
+        :linenos:
+
+.. note::
+    The ``[ForeignKey]`` annotation can be placed on either navigation property in the relationship. It does not need to go on the navigation property in the dependent entity class.
+
```

Comment:
`[ForeignKey]` can also be put on property and also putting on property essentially defines the dependent end.